client/src/pages/admin/Users.tsx

```diff
@@ -1,7 +1,664 @@
+  const { data: majorsData } = useQuery('get', '/v1/majors', {
+    options: {
+      refetchOnWindowFocus: false,
+      refetchOnMount: true,
+      refetchOnReconnect: false,
+      retry: 1,
+      staleTime: 30000
+    }
+  });
+
+  const { data: educationLevelData } = useQuery('get', '/v1/enums/{enumType}', {
+    params: {
+      path: {
+        enumType: 'education_level_enum'
+      }
+    },
+    options: {
+      refetchOnWindowFocus: false,
+      refetchOnMount: true,
+      refetchOnReconnect: false,
+      retry: 1,
+      staleTime: 30000
+    }
+  });
+
+  const { data: roleData } = useQuery('get', '/v1/enums/{enumType}', {
+    params: {
+      path: {
+        enumType: 'user_role_enum'
+      }
+    },
+    options: {
+      refetchOnWindowFocus: false,
+      refetchOnMount: true,
+      refetchOnReconnect: false,
+      retry: 1,
+      staleTime: 30000
+    }
+  });
+
+  const { data: membershipTermData } = useQuery('get', '/v1/enums/{enumType}', {
+    params: {
+      path: {
+        enumType: 'membership_term_enum'
+      }
+    },
+    options: {
+      refetchOnWindowFocus: false,
+      refetchOnMount: true,
+      refetchOnReconnect: false,
+      retry: 1,
+      staleTime: 30000
+    }
+  });
+
+  // Extract options from API responses with type assertions
+  const majorOptions = (majorsData as MajorsResponse)?.majors.map((major) => major.name) || [];
+  const educationLevelOptions = (educationLevelData as EnumResponse)?.types || [];
+  const roleOptions = (roleData as EnumResponse)?.types || [];
+  const paidOptions = (membershipTermData as EnumResponse)?.types || [];
+
+  // Reset to first page when filters change
+  useEffect(() => {
+    setCurrentPage(1);
+  }, [filters]);
+
+  // Create a wrapper function that converts array values into the correct format for the API
+  const createArrayParam = <T extends string>(values: T[]): T | T[] | undefined => {
+    return values.length > 0 ? values : undefined;
+  };
+
+  // Construct the query parameters
+  const queryParams = {
+    name: filters.name ? filters.name.toLowerCase() : undefined,
+    page: currentPage.toString(),
+    per_page: itemsPerPage.toString()
+  };
+
+  // Use the useQuery hook with type-safe parameters
+  const { data, isLoading, error } = useQuery('get', '/v1/users', {
+    params: {
+      query: {
+        ...queryParams,
+        'education_level[]': createArrayParam(filters.education_level),
+        'major[]': createArrayParam(filters.major),
+        'role[]': createArrayParam(filters.role),
+        'paid[]': createArrayParam(filters.paid)
+      }
+    },
+    options: {
+      refetchOnWindowFocus: false,
+      refetchOnMount: true,
+      refetchOnReconnect: false,
+      retry: 1,
+      staleTime: 30000
+    }
+  });
+
+  // Update totalPages when data changes
+  React.useEffect(() => {
+    if (data) {
+      setTotalPages(Math.ceil((data as UsersResponse).total / itemsPerPage));
+    }
+  }, [data, itemsPerPage]);
```

Comment:
not sure if you need a state for totalPages. why not make it a variable. if updates to data or itemsPerPage rerenders the page, then the totalPages variable should also be recalculated.
```suggestion
const totalPages = Math.ceil((data as UsersResponse).total / itemsPerPage)
```